jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -98,45 +96,45 @@ public void publish(BufferedInputStream bis, String fileName, ImporterFileType f
       } while (status == ParserStatus.COMPLETE);
    }
 
-   public void parsePayload(List<OdeMsgPayload> payloadList) {
+   public void parsePayload(List<OdeData> dataList) {
 
-      OdeMsgPayload msgPayload;
+      OdeData odeData;
       
+      OdeMsgPayload payload;
+      OdeLogMetadata metadata;
       if (fileParser instanceof DriverAlertFileParser){
-         msgPayload = new OdeDriverAlertPayload(((DriverAlertFileParser) fileParser).getAlert());
+        payload = new OdeDriverAlertPayload(((DriverAlertFileParser) fileParser).getAlert());
+        metadata = new OdeLogMetadata(payload);
+        odeData = new OdeDriverAlertData(metadata, payload);
       } else {
-         msgPayload = new OdeAsn1Payload(fileParser.getPayloadParser().getPayload());
+        payload = new OdeAsn1Payload(fileParser.getPayloadParser().getPayload());
+        metadata = new OdeLogMetadata(payload);
+        odeData = new OdeAsn1Data(metadata, payload);
       }
+      fileParser.updateMetadata(metadata);
 
-      payloadList.add(msgPayload);
+      dataList.add(odeData);
    }
 
-   public void publish(XmlUtils xmlUtils, List<OdeMsgPayload> payloadList) throws JsonProcessingException {
-     serialId.setBundleSize(payloadList.size());
-     for (OdeMsgPayload msgPayload : payloadList) {
-       OdeLogMetadata msgMetadata;
-       OdeData msgData;
+   public void publish(XmlUtils xmlUtils, List<OdeData> dataList) throws JsonProcessingException {
+     serialId.setBundleSize(dataList.size());
+     for (OdeData odeData : dataList) {
+       OdeLogMetadata msgMetadata = (OdeLogMetadata) odeData.getMetadata();
        
        if (fileParser instanceof DriverAlertFileParser){
           logger.debug("Publishing a driverAlert.");
 
-          msgMetadata = new OdeLogMetadata(msgPayload);
           msgMetadata.setSerialId(serialId);
 
-          OdeLogMetadataCreatorHelper.updateLogMetadata(msgMetadata, fileParser);
-          
-          msgData = new OdeDriverAlertData(msgMetadata, msgPayload);
-          publisher.publish(JsonUtils.toJson(msgData, false),
+          publisher.publish(JsonUtils.toJson(odeData, false),
              publisher.getOdeProperties().getKafkaTopicDriverAlertJson());
           serialId.increment();
        } else {
           if (fileParser instanceof BsmLogFileParser || 
```

Comment:
Looks like this whole if/else block does nothing now?

Reply:
Why do you say that? It publishes to different topics and adds encoding if data is to be decoded.